### PR TITLE
[PotentialFlow] Implementing stabilization term for the incompressible embedded formulation

### PR DESCRIPTION
**Pull request by marcnunezc**
Hi!

In this PR I add an extra term in the embedded element to improve the gradient peaks.
image
Where \xi is computed by averaging the gradient of the neighbour elements.
This is a before/after picture:
embedded_incompressible_embedded_incompressible_background1_naca0012_5 0_REMESH_0 004_naca0012_5 0_11

embedded_incompressible_embedded_incompressible_background1_naca0012_5 0_REMESH_0 004_naca0012_5 0_PENALTY_10 0_7

The new term depends on a penalty coefficient and makes the incompressible element non-linear. The term is only computed if a penalty coefficient is defined and it's different than 0.

### CODE REVIEW

File: applications/CompressiblePotentialFlowApplication/custom_elements/embedded_incompressible_potential_flow_element.cpp [hunk 1]
```diff
@@ -12,6 +12,7 @@
 #include "embedded_incompressible_potential_flow_element.h"
 #include "compressible_potential_flow_application_variables.h"
 #include "custom_utilities/potential_flow_utilities.h"
+#include "fluid_dynamics_application_variables.h"
```
Comment (marcnunezc) [inline applications/CompressiblePotentialFlowApplication/custom_elements/embedded_incompressible_potential_flow_element.cpp:15]: I am using the PENALTY_COEFFICIENT from the FluidApp here. #6318 should be merged first

**Review by inigolcanalejo — COMMENTED** [review]
The results look much better! Maybe a test for this case should be added?

**Comment by marcnunezc** [comment]
Sure!

**Review by inigolcanalejo — DISMISSED** [review]
Thanks! :+1:

**Comment by marcnunezc** [comment]
Hi @inigolcanalejo, I have made a small change, now I consider the elements touching the trailing edge. Earlier this was causing the solution to not converge, but the issue was that I was not computing correclty the gradient of the wake elements (not taking into account the auxiliary dofs). With this change it works fine now considering all elements
```

                Vector neighbour_elemental_gradient;
                if (is_neighbour_wake == 0) {
                    neighbour_elemental_gradient = PotentialFlowUtilities::ComputeVelocityNormalElement<Dim,NumNodes>(r_elem);
                }
                else {
                    neighbour_elemental_gradient = PotentialFlowUtilities::ComputeVelocityUpperWakeElement<Dim,NumNodes>(r_elem);
                }
```

**Review by inigolcanalejo — APPROVED** [review]
Sounds good!

**Pull request merged**